packages/profile-sync-controller/src/controllers/user-storage/network-syncing/controller-integration.ts

```diff
@@ -32,38 +64,141 @@ export function startNetworkSyncing(props: SetupNetworkSyncingProps) {
   } catch (e) {
     log.warn('NetworkSyncing, event subscription failed', e);
   }
+}
 
+/**
+ * Action to perform the main network sync.
+ * It will fetch local networks and remote networks, then determines which networks (local and remote) to add/update
+ * @param props - parameters used for this main sync
+ */
+export async function performMainNetworkSync(
+  props: PerformMainNetworkSyncProps,
+) {
+  const { messenger, getStorageConfig } = props;
+
+  // Edge-Case, we do not want to re-run the main-sync if it already is in progress
+  /* istanbul ignore if - this is not testable */
+  if (isMainNetworkSyncInProgress) {
+    return;
+  }
+
+  isMainNetworkSyncInProgress = true;
   try {
-    messenger.subscribe(
-      'NetworkController:networkDeleted',
-      // eslint-disable-next-line @typescript-eslint/no-misused-promises
-      async (networkConfiguration) => {
+    const opts = await getStorageConfig();
+    if (!opts) {
+      return;
+    }
+
+    const networkControllerState = messenger.call('NetworkController:getState');
+    const localNetworks = Object.values(
+      networkControllerState.networkConfigurationsByChainId ?? {},
+    );
+
+    const remoteNetworks = await getAllRemoteNetworks(opts);
+    const networkChanges = findNetworksToUpdate({
+      localNetworks,
+      remoteNetworks,
+    });
+
+    log.debug('performMainNetworkSync() - Network Syncing Started', {
```

Comment:
Debug logs get stripped, so I'm inclined to keep this as it is really helpful for debugging or testing...

But can remove.

Reply:
As in debug logs get stripped when in production:
https://github.com/MetaMask/metamask-extension/blob/8bcd777d0b060267d19ae1c3d25a36ab99f6ea87/app/scripts/background.js#L114

This is handy since we can create dev builds with the debug flag enabled (to help us introspect and investigate the decrypted user storage entries)